src/scripts/main.js

```diff
@@ -1,3 +1,352 @@
+      board[k][i] = newColumn[k];
+    }
+  }
+
+  checkAndAction(check);
+};
+
+const moveDown = (check) => {
+  for (let i = 0; i < board.length; i++) {
+    let newColumn = [];
+
+    for (let j = 0; j < board[0].length; j++) {
+      newColumn.push(board[j][i]);
+    }
+
+    newColumn = shiftRowLeft(newColumn).reverse();
+
+    for (let j = 0; j < newColumn.length; j++) {
+      if (newColumn[j - 1] === newColumn[j]) {
+        newColumn[j - 1] = 0;
+        newColumn[j] = newColumn[j] * 2;
+        score += newColumn[j];
+      }
+    }
+
+    newColumn = shiftRowLeft(newColumn).reverse();
+
+    for (let j = 0; j < newColumn.length; j++) {
+      board[j][i] = newColumn[j];
+    }
+  }
+
+  checkAndAction(check);
+};
+
+function changeClassCell(element, addClass) {
+  element.classList = '';
+  element.classList.add('field-cell');
+
+  if (addClass) {
+    element.classList.add(addClass);
+  }
+}
+
+function checkIfPlayerWon() {
+  const pointForVictory = 2048;
+
+  for (const row of board) {
+    for (const cell of row) {
+      if (cell === pointForVictory) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function handleVictory() {
+  messageWin.classList.remove('hidden');
+
+  start.textContent = 'Start';
+
+  start.classList
+    .remove('restart')
+    .add('start');
+
+  started = false;
+  messageStart.classList.remove('hidden');
+}
+
+if (checkAndAction) {
+  if (checkIfPlayerWon()) {
+    handleVictory();
+  }
+}
+
+function freeCell() {
+  for (const row of board) {
+    for (const cell of row) {
+      if (cell === 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function goToOldBoard(oldBoard, oldProbabilityCount, oldScore) {
+  board = [...oldBoard];
+  probabilityCount = oldProbabilityCount;
+  score = oldScore;
+}
+
+function checkIfPlayerLose() {
+  let lose = true;
+  const temp = [...board];
+  const tepmProbabilityCount = probabilityCount;
+  const tempScore = score;
+
+  moveLeft('check');
+
+  if (freeCell()) {
+    lose = false;
+    goToOldBoard(temp, tepmProbabilityCount, tempScore);
+
+    return lose;
+  }
+
+  moveUp('check');
+
+  if (freeCell()) {
+    lose = false;
+    goToOldBoard(temp, tepmProbabilityCount, tempScore);
+
+    return lose;
+  }
+
+  goToOldBoard(temp, tepmProbabilityCount, tempScore);
+
+  return lose;
+}
+
+function showDisplay() {
+  for (let i = 0; i < board.length; i++) {
+    const cells = fieldRowAll[i].querySelectorAll('.field-cell');
+
+    for (let j = 0; j < board[i].length; j++) {
+      const boardCell = board[i][j];
+      const cell = cells[j];
+
+      if (boardCell !== 0) {
+        cell.textContent = boardCell;
+
+        const addClass = `field-cell--${boardCell}`;
+
+        changeClassCell(cell, addClass);
+      } else {
+        cell.textContent = '';
+        changeClassCell(cell);
+      }
+    }
+  }
+
+  scoreDisplay.textContent = score;
+}
+
+function startGame() {
+  board = [
```

Comment:
You create the same board twice. Would be great to create some constant like a defaultBoard and just make copies then.